ui/src/widgets/ui-file-input/UIFileInput.vue

```diff
@@ -0,0 +1,161 @@
+    },
+    setup (props) {
+        useDataTracker(props.id)
+    },
+    data () {
+        return {
+            files: null,
+            uploading: false,
+            uploaded: false,
+            progress: 0
+        }
+    },
+    computed: {
+        ...mapState('data', ['messages']),
+        icon: function () {
+            const icon = this.props.icon
+            return icon ? 'mdi-' + icon.replace(/^mdi-/, '') : null
+        },
+        accept: function () {
+            return this.props.accept
+        },
+        label: function () {
+            return this.props.label
+        },
+        showFileSize: function () {
+            return this.props.showFileSize
+        },
+        multiple: function () {
+            return this.props.multiple
+        }
+    },
+    methods: {
+        formatFileSize (bytes) {
+            if (bytes === 0) return '0 Bytes'
+            const k = 1000
+            const dm = 2
+            const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB']
+            const i = Math.floor(Math.log(bytes) / Math.log(k))
+            return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i]
+        },
+        reset: function () {
+            this.files = null
+            this.uploading = false
+            this.uploaded = false
+            this.progress = 0
+        },
+        upload: function (file) {
+            if (file && !this.multiple) {
+                // Create a FileReader instance to read the file
+                const reader = new FileReader()
+
+                // When the file is read, send it to Node-RED
+                reader.onload = () => {
+                    // Prepare the payload to send
+                    const msg = {
+                        payload: file, // File content
+                        file: {
+                            name: file.name, // File name
+                            size: file.size, // File size
+                            type: file.type // File type
+                        }
+                    }
+
+                    this.uploading = false
+                    this.uploaded = true
+
+                    this.send(msg)
+                }
+
+                // Track progress of file reading
+                reader.onprogress = (event) => {
+                    this.progress = event.loaded // Update progress
+                }
+                this.uploading = true
+
+                // readAsText alternative?
+                reader.readAsArrayBuffer(file)
+            }
+        },
+        send (msg) {
+            this.$socket.emit('widget-send', this.id, msg)
+        },
+        maxFileSize (files) {
+            if (files) {
+                const size = files[0]?.size
+                const maxSize = this.props.maxFileSize
+                return !files || !files.length || size < maxSize || `File size should be less than ${this.formatFileSize(maxSize)}!`
+            }
```

Comment:
doesnt compute all files in multiple selection

Reply:
Addressed in https://github.com/FlowFuse/node-red-dashboard/pull/981/commits/1fe18a05881f60fd743016f375281a02e8f3cb81